### ISSUE TITLE
Path regular expression is inconsistent to expectations for repeated "any predicate"

### ISSUE THREAD

**Issue by hoijnet** [terminusdb/terminusdb#347]
**Describe the bug**
A path regular expression "(*)+" or "(*){1,2} does not identify/traverse more than one level from source to target which would be expected considering that "(property_name_level_1|property_name_level_2)+" does traverse more than one level.

**To Reproduce**
1. Go to TerminusDB Console Query tab
2. Enter any of the following:
   * Traverses two levels: ```WOQL.path("v:source", "(propertyname_firstlevel|propertyname_secondlevel){1,2}", "v:target", "v:traversed")```
   * Does not work, expecting same traversals or more than above: ```WOQL.path("v:source", "(*){1,2}", "v:target", "v:traversed")```
   * Traverses two levels: ```WOQL.path("v:source", "(propertyname_firstlevel|propertyname_secondlevel)+", "v:target", "v:traversed")```
   * Does not work, expecting same traversals or more than above: ```WOQL.path("v:source", "(*)+", "v:target", "v:traversed")```
3. Click Run Query

**Expected behavior**
For the ones that work I got the correct result, traversal from ```source-->intermediary-->target``` and the path in v:traversed.

For the ones that does not work, when using "any predicate, *", only one level is traversed: ```source-->intermediary```, and only the path for the source-->intermediary traversal is present in the result.

**Screenshots**
N/A

**Info (please complete the following information):**

![image](https://user-images.githubusercontent.com/144317/123678461-9db30e80-d846-11eb-8361-d0bad7ed9c94.png)

**Additional context**
N/A, small database, minimal test does not yield expected behaviour

**Comment by spl** [comment]
Thanks for this, too. We're currently working on revamping a lot of things at the moment. But we'll look into this problem soon.

**Comment by GavinMendelGleason** [comment]
Yes, actually this *should* be solved by #755 but the syntax will be to use '.' rather than '*' which is reserved for pattern repetition multiple times (as in Regex). It is not yet in the JS or Python client yet, but should be by next week after we upgrade the TerminusX servers.

**Comment by hoijnet** [comment]
Looking forward to trying it out!
